enzy_htp/_interface/pymol_interface.py

```diff
@@ -101,6 +101,86 @@ def select_pymol_obj(self, pattern: str, pymol_obj_name: str,
+        Args:
+            pos_key: the chain id and residue index.
+            target: the target residue name (3 letters).
+            pymol_obj_name: the name of the target WT in PyMOL.
+            pymol_session: the target PyMOL session.
+            debug: prints all rotamer scores out for debugging purposes.
+        Returns:
+            None.
+        """
+
+        # load pymol wizard mutagenesis function
+        pymol_session.cmd.wizard("mutagenesis")
+        pymol_session.cmd.do("refresh_wizard")
+        
+        # select res idx to mutate to target
+        pymol_session.cmd.get_wizard().set_mode(target)
+        if pymol_obj_name == "":
+            pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+        else:
+            pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+
+        # prints all rotamers and strains out; also saves each variation to a PDB file in scratch/.
+        # can use for debugging purposes
+        if debug:
+            for i in range(1, pymol_session.cmd.count_states() + 1):
+                pymol_session.cmd.wizard("mutagenesis")
+                pymol_session.cmd.do("refresh_wizard")
+                pymol_session.cmd.get_wizard().set_mode(target)
+                if pymol_obj_name == "":
+                    pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+                else:
+                    pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+            
+                pymol_session.cmd.get_wizard().do_state(i)
+                pymol_session.cmd.frame(i)
+                pymol_session.cmd.get_wizard().apply()
+                self.save_to_stru(pymol_obj_name, pymol_session, True)
+            return
+
+        pymol_session.cmd.get_wizard().apply()
+
+
+    def save_to_stru(self, pymol_obj_name: str,
+                    pymol_session: pymol2.PyMOL,
+                    in_order: bool = True,
+                    keep_files = False) -> Structure:
+        """
+        Saves a PyMOL object to a Structure object.
+        Args:
+            pymol_obj_name: the name of the target enzyme in PyMOL.
+            pymol_session: the target PyMOL session.
+            in_order: if the saving should keep the order of the atoms in the original object.
+            keep_files: if the program should clean up the temp files or not.
+        Returns:
+            A Structure object representing the target enzyme in PyMOL.
+        """
+        sp = PDBParser()
+        temp_dir = eh_config["system.SCRATCH_DIR"]
+        fs.safe_mkdir(temp_dir)
+        pymol_outfile_path = fs.get_valid_temp_name(
+                f"{eh_config['system.SCRATCH_DIR']}/pymol_output.pdb")
+
+        if in_order:
+            pymol_session.cmd.set("retain_order")
+
+        pymol_session.cmd.select(pymol_obj_name)
+        pymol_session.cmd.save(pymol_outfile_path, "(sele)")
```

Comment:
Would it be different from just use the pymol_obj_name instead of (sele) and avoid the redundant cmd.select?